File: 069_sqrtx/Solution.py
class Solution:
    def mySqrt(self, x: int) -> int:
        l, r, res = 0, x, -1

        while l <= r:
            mid = (l + r) // 2
            if mid * mid <= x:
                res = mid
                l = mid + 1
            else:
                r = mid - 1

        return res


if __name__ == '__main__':
    print(Solution().mySqrt(5))

File: 069_sqrtx/Solution.java
class Solution {
    public int mySqrt(int x) {
        int l = 0, r = x, res = -1;
        while (l <= r) {
            int mid = l + (r - l) / 2;
            if ((long) mid * mid <= x) {
                res = mid;
                l = mid + 1;
            } else {
                r = mid - 1;
            }
        }

        return res;
    }

    public static void main(String[] args) {
        System.out.println(new Solution().mySqrt(5));
    }
}
